packages/manager/modules/product-offers/src/workflows/product-offers-workflow.class.js

```diff
@@ -0,0 +1,202 @@
+import isEqual from 'lodash/isEqual';
+import some from 'lodash/some';
+import sortBy from 'lodash/sortBy';
+
+import Pricing from '../pricing/pricing.class';
+import ProductOffersService from '../services/product-offers.service';
+import { PRICING_CAPACITIES } from '../pricing/pricing.constants';
+
+/**
+ * Workflow base class
+ * Each type of workflow (see WORKFLOW_TYPES), inherits this abstract class
+ * @abstract
+ */
+export default class Workflow {
+  /**
+   * @param {Object} $translate      AngularJS provider
+   * @param {Object} workflowOptions Workflow specific options. Each type of
+   * workflow has its own options to work with.
+   * See other workflow class constructor documentation for more details.
+   */
+  constructor($q, $translate, workflowOptions) {
+    this.$q = $q;
+    this.$translate = $translate;
+
+    if (new.target === Workflow) {
+      throw new TypeError(
+        'ovhProductOffers-Workflow: Workflow cannot be instanciated directly',
+      );
+    }
+
+    if (!workflowOptions) {
+      throw new Error(
+        'ovhProductOffers-Workflow: workflowOptions is undefined',
+      );
+    }
+
+    Object.assign(this, workflowOptions);
```

Comment:
Maybe `workflowOptions` should be destructured so we can know which property can be expected

Reply:
There is documentation for each type of `workflowOptions`